ambry-api/src/main/java/com.github.ambry/router/Router.java

```diff
@@ -57,6 +66,16 @@
   public Future<ReadableStreamChannel> getBlob(String blobId, Callback<ReadableStreamChannel> callback);
 
   /**
+   * Requests for the blob data asynchronously and invokes the {@link Callback} when the request completes.
+   * @param blobId The ID of the blob for which blob data is requested.
+   * @param options The options associated with the request.
+   * @param callback The callback which will be invoked on the completion of the request.
+   * @return A future that would contain a {@link ReadableStreamChannel} that represents the blob data eventually.
+   */
+  public Future<ReadableStreamChannel> getBlob(String blobId, GetBlobOptions options,
```

Comment:
It is not very clear from the comments how this method is different from the other getBlob apis?